pkg/agent/replica_set_port_manager.go

```diff
@@ -0,0 +1,108 @@
+package agent
+
+import (
+	"github.com/mongodb/mongodb-kubernetes-operator/pkg/automationconfig"
+	"go.uber.org/zap"
+	corev1 "k8s.io/api/core/v1"
+)
+
+type ReplicaSetPortManager struct {
+	log              *zap.SugaredLogger
+	expectedPort     int
+	currentPodStates []PodState
+	currentAC        automationconfig.AutomationConfig
```

Comment:
Do we really need to store the entire automation config here, can be just store the processes which is needed I guess?

Reply:
Good catch. We need only processes!

Reply:
Done